catfs/mio/overlay/overlay.go

```diff
@@ -228,6 +228,24 @@ func (l *Layer) Write(buf []byte) (int, error) {
 	return len(buf), nil
 }
 
+// Writes data from `buf` at offset `off` counted from the start (0 offset).
+// Mimics `WriteAt` from `io` package https://golang.org/pkg/io/#WriterAt
+func (l *Layer) WriteAt(buf []byte, off int64) (n int, err error) {
+	// Copy the buffer, since we cannot rely on it being valid forever.
+	modBuf := make([]byte, len(buf))
```

Comment:
That documentation is a little misleading. Nothing gets garbage collected while you still use it (i.e. you hold a reference to it). ``modBuf`` can be ``Put()`` back to the pool after flush and on L235 above you can do a ``pool.Get()`` to fetch a buffer that was either used before or is newly allocated. Go is free to do whatever it likes in the time between Put and Get.

Reply:
I see now. Can we move it to separate issue?
 From quick reading, `pool` is not suited for different size objects. So if `len(buf)` different from result of a `Get`, we would have to put it back. But I see no way, to iterate over `pool` to see if there is a suitable candidate. Besides, it would require the full rethinking of how `overlay` is working.

Reply:
You would just get an item from the pool and check if it's reasonably sized (if larger, cut it), if not allocate a new one the normal way and use that. But sure, we can do it once overlay is touched in general.